sessionmanager/sessionmanager.go

```diff
@@ -0,0 +1,65 @@
+package sessionmanager
+
+import (
+	"sync"
+
+	exchange "github.com/ipfs/go-ipfs-exchange-interface"
+)
+
+type SessionManager interface {
+	GetNextSessionID() uint64
+	IterateSessions(iterate iterateSessionFunc)
+	AddSession(session exchange.Fetcher)
+	RemoveSession(session exchange.Fetcher)
+}
+
+type sessionManager struct {
+	// Sessions
+	sessions []exchange.Fetcher
```

Comment:
Please use the [mutex hat](https://dmitri.shuralyov.com/idiomatic-go#mutex-hat) pattern to make clear what the mutex is protecting.

Reply:
Thanks for that link. I'm relatively new to go and was looking for resource on what's considered "idiomatic" in go.